src/transformers/models/bert/modeling_bert.py

```diff
@@ -192,10 +203,22 @@ def forward(
         seq_length = input_shape[1]
 
         if position_ids is None:
-            position_ids = self.position_ids[:, past_key_values_length : seq_length + past_key_values_length]
+            if hasattr(self, "position_ids"):
+                position_ids = self.position_ids[:, past_key_values_length : seq_length + past_key_values_length]
+            else:
+                position_ids = torch.arange(seq_length).expand((1, -1))
 
         if token_type_ids is None:
-            token_type_ids = torch.zeros(input_shape, dtype=torch.long, device=self.position_ids.device)
+            if hasattr(self, "token_type_ids"):
+                token_type_ids = self.token_type_ids[:, :seq_length]
```

Comment:
If using the `BertEncoder` as a standalone, this will work. If using any other architecture, like `BertModel` or `BertForXXX`, by now the token type IDs have already been created. Their creation happens earlier:

https://github.com/huggingface/transformers/blob/d2753dcbec7123500c1a84a7c2143a79e74df48f/src/transformers/models/bert/modeling_bert.py#L939-L940

You could manually check there `if hasattr(self.embeddings, "token_type_ids"):` and act accordingly.

Reply:
If `self` doesn't have `token_type_ids` as an attribute, then you need to resort to the way it was previously handled

Reply:
```suggestion
            if hasattr(self, "token_type_ids"):
                token_type_ids = self.token_type_ids[:, :seq_length]
            else:
                token_type_ids = torch.zeros(input_shape, dtype=torch.long, device=self.position_ids.device)
```

Reply:
This will craft a tensor of dimension `[1, seq_len]`, when it was previously `[batch_size, seq_len]`

Reply:
@LysandreJik , need to change token_type_ids = self.token_type_ids[:, :seq_length] to token_type_ids = self.token_type_ids[input_shape[0], :seq_length]? will update.